sig-security-assessments/cluster-api/self-assessment.md

```diff
@@ -0,0 +1,1099 @@
+  * Checks for breaking API changes
+  * Tests for broken URLs in generated documentation
+  * PRs cannot be merged without passing tests without an explicit /override
+    from a maintainer
+* Each PR requires approval from maintainers listed in the OWNERS file
+* For all Cluster API subprojects, contributors are required to have signed the
+  CNCF CLA, but no cryptographic signatures are required for individual commits.
+* Pushes to the main branch are forbidden by convention
+* Release process is almost fully automated via Github actions
+* Checksum, signature and SPDX files are not currently generated for any
+  artifacts
+* Container images are produced automatically by Google CloudBuild, and Cluster
+  API uses the upstream Kubernetes image promotion process to promote individual
+  OCI SHAs from staging to release.
+* Deployment manifests reference image tags rather than SHA digests
+* ACLs for ownership of the repository and approvals are configured in code
+  within the OWNERS file at the top of the repository as well as configured via
+  the github.com/kubernetes/org repository.
+* For Cluster API Provider AWS since v0.6.7 and Cluster API Provider Azure since
+  v0.4.13 maintainers do GPG sign all tags, and maintainers are expected to
+  maintain their GPG keys with public keys verifiable in Github. Cluster API
+  Core repository has tags signed by Github Actions since v1.0.3
+
+#### Identified gaps in development workflow
+
+##### [DEV-01: Outdated OWNERS file](https://github.com/kubernetes-sigs/cluster-api/issues/6151)
+
+Listed maintainers of the Cluster API project are out of date and may continue
+to have administrative access to the GitHub repository as well as perform image
+promotion and access the staging Google Cloud account. A process to regularly
+review ACLs(Access Control Lists) should be instituted.
+
+Status: implemented / planned / to be implemented
+
+#### Core Infrastructure Initiative
+
+Cluster API asserts being 96% in progress with regards to
+the [Core Infrastructure Initiative (CII) badging process](https://bestpractices.coreinfrastructure.org/en/projects/5288)
+.
+
+##### DEV-02: Vulnerability report process is poorly defined
+
+The project has a SECURITY_CONTACTS file which nominally follows the Kubernetes
+vulnerability reporting process, but the file has not been maintained, and has
+not been invoked for subprojects as far as the authors are aware. In addition,
+the contacts only lists GitHub usernames - it isn’t clear how Kubernetes
+security is to get in touch with project maintainers in the case of private
+disclosure. To fix this, we would like to request a private google group, like
+this `cluster-api-security@kubernetes.io` that can be used for private
+discussions with SRC on any potential vulnerabilities discovered or reported for
+cluster-api.
+
+Status: [Planned](https://github.com/kubernetes-sigs/cluster-api/issues/5398)
+
+##### DEV-03: Project should configure some fuzzing
```

Comment:
AFAIK done, but I have asked for updates on the issue

Reply:
Now closed: https://github.com/kubernetes-sigs/cluster-api/issues/6059